streaming/src/main/scala/org/apache/spark/streaming/util/HdfsUtils.scala

```diff
@@ -27,7 +27,7 @@ private[streaming] object HdfsUtils {
     // If the file exists and we have append support, append instead of creating a new file
     val stream: FSDataOutputStream = {
       if (dfs.isFile(dfsPath)) {
-        if (conf.getBoolean("hdfs.append.support", false)) {
+        if (conf.getBoolean("hdfs.append.support", false) || dfs.isInstanceOf[RawLocalFileSystem]) {
```

Comment:
@harishreedharan This change is to enable append when running locally. Without this the unit tests fail because all the units are running on manual clock and so all of them (when they reuse the same checkpoint directory) try to write to log files with the same name.